### ISSUE TITLE
bug: syntax error near unexpected token `elif'

### ISSUE THREAD

**Issue by lackovic** [asdf-vm/asdf#1297]
### Describe the Bug

On Ubuntu 22.04 WSL2, following the [3. Install asdf](https://asdf-vm.com/guide/getting-started.html#_3-install-asdf) instructions I get the error `syntax error near unexpected token `elif'`. May be related to https://github.com/asdf-vm/asdf/issues/1187.

### Steps to Reproduce

Add the following two lines to your `~/.bashrc`:
```
. $HOME/.asdf/asdf.sh
. $HOME/.asdf/completions/asdf.bash
```
Then try to source the file:
```
$ source ~/.bashrc
```

### Expected Behaviour

It should source the file without errors.

### Actual Behaviour

The following errors are displayed:
```
-bash: /home/marco/.asdf/asdf.sh: line 9: syntax error near unexpected token `elif'
'bash: /home/marco/.asdf/asdf.sh: line 9: `  elif [[ "$_under" == *".sh" ]]; then
-bash: /home/marco/.asdf/completions/asdf.bash: line 1: syntax error near unexpected token `$'{\r''
'bash: /home/marco/.asdf/completions/asdf.bash: line 1: `_asdf() {
```

### Environment

```shell
Ubuntu 22.04 WSL2

$ asdf info
Command 'asdf' not found, did you mean:
  command 'sdf' from deb sdf (2.001+1-8)
  command 'adsf' from deb ruby-adsf (1.4.5+dfsg1-2)
  command 'asdfg' from deb aoeui (1.7+20160302.git4e5dee9-3)
  command 'sadf' from deb sysstat (12.5.2-2build2)
Try: sudo apt install <deb name>
```
```


### asdf plugins affected (if relevant)

_No response_

**Comment by jthegedus** [comment]
Copied from my comments on https://github.com/asdf-vm/asdf/issues/1187

---

So two things to keep in mind when using WSL2 environment:

From our [FAQ](https://asdf-vm.com/learn-more/faq.html)
>Importantly, WSL2 is only expected to work properly when the current working directory is a Unix drive and not a bound Windows drive.

The error reported shows:
```
bash: /home/USER/.asdf/asdf.sh: line 9: syntax error near unexpected token `elif'
'ash: /home/USER/.asdf/asdf.sh: line 9: `  elif [[ "$_under" == *".sh" ]]; then
bash: /home/USER/.asdf/completions/asdf.bash: line 1: syntax error near unexpected token `$'{\r''
'ash: /home/USER/.asdf/completions/asdf.bash: line 1: `_asdf() {
```
The 3rd line here shows an unexpected token `\r` which is likely due to this repository being cloned to a non-Unix location or by using Git in a Windows environment or by the line endings using Windows instead of *Nix. Changing your line endings to LF from CRLF might fix this.

As stated on the other threads, if this was an issue with the main entrypoint to asdf having a typo or illegal chars then there would be hundreds of people reporting this issue.

**Comment by lackovic** [comment]
> The 3rd line here shows an unexpected token \r which is likely due to this repository being cloned to a non-Unix location or by using Git in a Windows environment or by the line endings using Windows instead of *Nix. Changing your line endings to LF from CRLF might fix this.

In my case I was using a Unix location, using Git in Unix environment. To handle the endings automatically I set `git config --global core.autocrlf input` (see [Customizing git configuration](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration)) but last time I ran it I used `true` by mistake instead of `input` for my wrong interpretation of the documentation and that changed all the endings from LF to CRLF when I cloned the repo. 

Once I set the `autocrlf` to `input` and re-cloned the repo `asdf info` worked as expected.
